server/lib/activity-event.js

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (event, data, request, queryParams) {
```

Comment:
I was confused by the signature here momentarily, `queryParams` implies the query parameter data to use, yet you grab that out out of `request.query`. Can you rename this to `allowedQueryParams`?